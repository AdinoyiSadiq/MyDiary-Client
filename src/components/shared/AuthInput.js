import React from 'react';

const AuthInput = (props) => {
  const { name, type, placeholder, id, value, handleChange, handleBlur, error, touched } = props;

  return (
    <div className="authWrapper">
      <input
        type={type || 'text'}
        placeholder={placeholder}
        id={id || name}
        value={value}
        onChange={handleChange}
        onBlur={handleBlur}
      />
      <div className="error">{touched[name] ? error[name] : ''}</div>
    </div>
  );
};

export default AuthInput;