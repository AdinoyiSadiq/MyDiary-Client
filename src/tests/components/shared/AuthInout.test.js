import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import AuthInput from '../../../components/shared/AuthInput';

let wrapped;

const touched = {
  name: 'name'
}

beforeEach(() => {
  wrapped = shallow(<AuthInput touched={touched} />);
});


describe('AuthInput UI', () => {
  describe('render features', () => {
    test('view should render as expected', () => {
      const tree = toJson(wrapped);
      expect(tree).toMatchSnapshot();
    });
  });
});