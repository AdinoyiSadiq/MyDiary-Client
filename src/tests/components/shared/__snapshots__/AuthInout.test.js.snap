// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthInput UI render features view should render as expected 1`] = `
<div
  className="authWrapper"
>
  <input
    type="text"
  />
  <div
    className="error"
  />
</div>
`;
